### PR TITLE
Fix/#160-display-speed

### LINKED ISSUE #160: Speed of values being displayed may be not realistic?
**Actual behavior**
This came up when considering the unit of the time-sweep indicator (#151). But is has to do with time-sweep only indirectly. If this is an issue, it does not come from the time-sweep itseld but more from the overall way of how data is processed and displayed in the frontend.

**Expected behavior**
Other oscilloscopes might (as I understood) display new data in their own speed, meaning a new div (pixel, respectively) in a fixed amount of time (unit: [[time/div](https://www.virtuelles-oszilloskop.de/time_div_wheel/)]). Also, Wieland said that our time-sweep should be unit time/div. But currently that is not what our oscilloscopse (and by this also our time-sweep) does.

**Additional context**
Changing this would require changing the whole way of how data is processed in the frontend (I guess).

### PR DESCRIPTION
**Pull request by nicolaskolbenschlag**
…weep computation to seperate helper so that it can be reused by slider's display-value

Signed-off-by: nicolaskolbenschlag <nicolas.kolbenschlag@googlemail.com>

### CODE REVIEW

**Comment by nicolaskolbenschlag** [comment]
For an explanation on the current implementation of this branch, please refer to the comments of the issue #160.

**Review by PhlppKrmr — COMMENTED** [review]
Like the code structure, that part is fine. Sadly I have a hard time reviewing the functionality cause it is pretty inter-dependent with the WebGL-migration and also the sending frequency of the generator. I see that we expect 10_000 updates per second, but we have no real way of ensuring that.

**Comment by nicolaskolbenschlag** [comment]
> Like the code structure, that part is fine. Sadly I have a hard time reviewing the functionality cause it is pretty inter-dependent with the WebGL-migration and also the sending frequency of the generator. I see that we expect 10_000 updates per second, but we have no real way of ensuring that.

It is true that we have no way of ensuring that. This [comment](https://github.com/amosproj/amos2022ws03-software-oscilloscope/issues/160#issuecomment-1397448770) explains why we expect this.

**Comment by nicolaskolbenschlag** [comment]
> Like the code structure, that part is fine. Sadly I have a hard time reviewing the functionality cause it is pretty inter-dependent with the WebGL-migration and also the sending frequency of the generator. I see that we expect 10_000 updates per second, but we have no real way of ensuring that.

Regarding the wegGL dependency: this code mainly works on the display value of the sliders. I think it should not be inter-dependent with the WegGL migration.

File: Apps/frontend/src/const.js [hunk 1]
```diff
@@ -52,3 +52,10 @@ export const BACKEND_BASE_URL = import.meta.env.VITE_BACKEND_BASE_URL;
 export const BACKEND_API_URL = `${BACKEND_BASE_URL}`;
 export const REST_ENDPOINT_CONFIG = `${BACKEND_API_URL}/config`;
 /* END REST API CONFIG */
+
+/* DISPLAY SPEED */
+const EXPECTED_UPDATES_PER_SECOND = 10_000;
```
Comment (jenswaechtler) [inline Apps/frontend/src/const.js:57]: I'm not a fan of this, but due to the short time until the end of the project, I think it's reasonable that we assume a fixed frequency.

**Review by jenswaechtler — CHANGES_REQUESTED** [review]
Good work.
Can you please run the linter over it again? 
When I run `npm run format`, it still changes several files for me. Including the ones you have touched.

File: Apps/frontend/src/views/ControlPanelBottom.svelte [hunk 2]
```diff
@@ -9,15 +9,22 @@
...
     expandedPanelOpen,
     offsetAdjustment,
     timeSweep,
   } from "../stores";
+  import { computeDisplayDeltaFromTimeSweep } from "../helper";
 
   export let controlPanelBottomHeight = 0;
+
+  const computeDisplaySpeed = (value) => {
+    let delta = computeDisplayDeltaFromTimeSweep(value);
```
Comment (motschel123) [inline Apps/frontend/src/views/ControlPanelBottom.svelte:25]: Not a big fan of splitting this into two similar named functions, but fine if it gets the jobs done.

**Review by motschel123 — APPROVED** [review]
Thanks for clarifying the issue again (discord) did lose track of what the problem was.

File: Apps/frontend/src/const.js (hunk 1 above)
Comment (leon-juenemann) [reply Apps/frontend/src/const.js:57]: As a next step, the backend can measure the number of packages received from the generator and notify the front end about this. I am pretty sure a real digital oscilloscope would also need to know about the sampling rate in order to plot the signal with the set time/div value.

**Review by leon-juenemann — APPROVED** [review]

**Review by jenswaechtler — APPROVED** [review]

**Comment by nicolaskolbenschlag** [comment]
The changes that were reviewed over this PR were merged via another PR due to conficts.

**Pull request closed without merging**